Python/sysmodule.c

```diff
@@ -1941,6 +1941,37 @@ sys__clear_type_cache_impl(PyObject *module)
     Py_RETURN_NONE;
 }
 
+#ifdef Py_DEBUG
+/*[clinic input]
+sys._get_type_version_tag
+
+    type: object
+    /
+
+For internal use only: get the `tp_version_tag` of *type*.
+[clinic start generated code]*/
+
+static PyObject *
+sys__get_type_version_tag(PyObject *module, PyObject *type)
+/*[clinic end generated code: output=bdcfab8b295e0a76 input=6016e5a02b3c3675]*/
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    _Py_EnsureTstateNotNULL(tstate);
+
+    if (!PyType_Check(type)) {
+        _PyErr_SetString(tstate, PyExc_TypeError, "argument must be a type");
+        return NULL;
+    }
+    PyObject *res = PyLong_FromUnsignedLong(
+        (unsigned long)((PyTypeObject *)type)->tp_version_tag);
```

Comment:
Implicit cast from unsigned int to unsigned long (same or larger type) is safe. No need to cast explicitly.

```suggestion
        ((PyTypeObject *)type)->tp_version_tag);
```